SMLHelper/Handlers/IngameMenuHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿namespace SMLHelper.V2.Handlers
+{
+    using System;
+    using SMLHelper.V2.Interfaces;
+    using SMLHelper.V2.Patchers;
+
+    /// <summary>
+    /// A handler class that offers simple ways to tap into functionality of the in game menu.
+    /// </summary>
+    public class IngameMenuHandler : IIngameMenuHandler
```

Comment:
This could probably include a function to block the saving of the game. I can provide code if you want.

Reply:
If that would be useful, then sure.  
I would suggest making these into method calls (not properties).  
Something like this:
```CS
void BlockSaving();

void AllowSaving();
```
And yes, if you can provide the code, go right ahead.

Reply:
I would suggest an implementation like this:

```cs
void RegisterSaveBlockingEvent(Func<bool> function);
```

with an usage like this:

```cs
RegisterSaveBlockingEvent(() =>
{
	bool b = false;
    // do something with b
    return b; // true to block, false to not
});

Reply:
That could also work.  
I can see both of these approaches having their use cases.

If you're able to code for both of them, that would be ideal.

Reply:
Where would the modder call `BlockSaving` and `AllowSaving`?

Reply:
At the beginning and end of some event she saving should be blocked.

Or the start and end of some animation sequence.
Ask @MrPurple641 about changing the outcome of the Sunbeam

Reply:
I've worked with save blocking in the past.

> At the beginning and end of some event she saving should be blocked.

Reply:
> So, SMLHelper would keep track of assemblies that called that function?
It would be up to the individual mods to re-enable saving if they disable it.

If a modder forgets to call `AllowSaving` at some point after they called `BlockSaving`, it would be no different than if their blocking function always returned true and never changed.

Reply:
I meant, how would SMLHelper differentiate between two mods calling `BlockSaving` and `AllowSaving` for their own reasons?

Reply:
Get the calling assembly
```CS
Assembly mod = ReflectionHelper.CallingAssemblyByStackTrace();

Blockers ICollection<Assembly>

BlockSaving() => Blockers.Add(callingAssembly);
AllowSaving() => Blockers.Remove(callingAssembly);

If (Blockers.Count > 0)
// block saving
```

Reply:
```cs
[HarmonyPatch(typeof(IngameMenu))]
[HarmonyPatch("GetAllowSaving")]
static class IngameMenu_GetAllowSaving
{
    static void Postfix(ref bool __result)
    {
        __result = __result && !Mod.disableSaving;
    }
}
```

Reply:
@AlexejheroYTB  The "block game from saving" feature will have to wait for a later update.  
It won't be included as part of this branch's release.